readthedocs/doc_builder/environments.py

```diff
@@ -931,7 +931,7 @@ def get_client(self):
             # Instead, give the user a generic failure
             raise BuildEnvironmentError(
                 BuildEnvironmentError.GENERIC_WITH_BUILD_ID.format(
-                    build_id=self.build['id'],
+                    build_id=self.build.get('id') if self.build else '',
```

Comment:
I feel like this whole block should be conditional on whether or not this is a build operation vs sync operation. Is there any UI to even show this error if raised in a sync operation?

Reply:
Nope. Yeah, we can just put a conditional